data/linux-bridge/002-linux-bridge.yaml

```diff
@@ -39,23 +39,23 @@ spec:
             - /bin/bash
             - -ce
             - |
-              echo 'Installing bridge and tuning CNIs'
               cni_mount_dir=/opt/cni/bin
-              sourcebinpath=/usr/src/github.com/containernetworking/plugins/bin
-              cp --remove-destination ${sourcebinpath}/bridge ${cni_mount_dir}/cnv-bridge
-              cp --remove-destination ${sourcebinpath}/tuning ${cni_mount_dir}/cnv-tuning
+              if find ${cni_mount_dir}/bridge &>/dev/null && ! test -L ${cni_mount_dir}/bridge; then
+                ln -s ${cni_mount_dir}/bridge ${cni_mount_dir}/cnv-bridge
+                ln -s ${cni_mount_dir}/tuning ${cni_mount_dir}/cnv-tuning
+              else
```

Comment:
we used to have conditional soft link creation
`find ${cni_mount_dir}/bridge &>/dev/null || ln -s ${cni_mount_dir}/cnv-bridge ${cni_mount_dir}/bridge`
but now we don't

and even on U/S we have already binaries `bridge`, `tuning`, that are installed
as part of provision by this command
```
dnf install --skip-broken --nobest --nogpgcheck --disableexcludes=kubernetes -y \
    kubeadm-${version} \
    kubelet-${version} \
    kubectl-${version} \
    kubernetes-cni
```
(note that removing `kubernetes-cni` wont help, it will still be installed)

It means that until today on U/S, we had bridge that belong to k8s and cnv-bridge that was installed by us.

On OCP it will be fine, but we need to decide please what to do on U/S

Maybe we are fine to keep U/S and D/S the same in case `bridge` binary exists ? (best imho if covers every need)
or we want to try and delete those two binaries, during cluster-up of CNAO (but it is prone to be less predictable)

Reply:
If we use the OpenShift strategy on Kubernetes, we may damage existing users who don't deploy Kubernetes from the RPMs you shared above. We need to keep shipping and updating our binaries, at least for now.

What about this logic:

```
if cnv-bridge symlink exists:
    do nothing
elif bridge binary exists:
   create cnv-bridge symlink
else:
  install cnv-bridge binary
  create bridge symlink
```

This way we should:
* Maintain the existing behavior on Kubernetes
* Always symlink to `bridge` binary if it is not installed by us
* Keep updating the binary if it is deployed by us
* Not need any additional files or flags

Is there any gap in my logic? @oshoval

Reply:
Thanks

Few notes to consider please (maybe it is fine, just raising)

It won't keep the current logic on kubevirtci
because bridge exists so it will "create cnv-bridge symlink"
before this PR, we did install cnv-network binary, but kept bridge binary as is

"If we use the OpenShift strategy on Kubernetes, we may damage existing users who don't deploy Kubernetes from the RPMs you shared above"
if they don't deploy it and don't have bridge, we install cnv-network for them and then create bridge softlink
isn't it fine ?

beside that sounds good

Reply:
Here is the code, the only change is the first if
i feel it isn't adding anything major comparing to current PR state, as it is already encapsulated in the 2nd condition ?
```
              cni_mount_dir=/opt/cni/bin
              if find ${cni_mount_dir}/cnv-bridge &>/dev/null && test -L ${cni_mount_dir}/cnv-bridge; then
                echo "cnv-bridge symbolic link found, nothing to be done"
              if find ${cni_mount_dir}/bridge &>/dev/null && ! test -L ${cni_mount_dir}/bridge; then
                echo "bridge binary found, creating cnv-bridge symbolic link"
                ln -sf ${cni_mount_dir}/bridge ${cni_mount_dir}/cnv-bridge
                ln -sf ${cni_mount_dir}/tuning ${cni_mount_dir}/cnv-tuning
              else
                echo "installing cnv-bridge and creating bridge symbolic link"
                sourcebinpath=/usr/src/github.com/containernetworking/plugins/bin
                cp --remove-destination ${sourcebinpath}/bridge ${cni_mount_dir}/cnv-bridge
                cp --remove-destination ${sourcebinpath}/tuning ${cni_mount_dir}/cnv-tuning

                printf -v bridgechecksum "%s" "$(<$sourcebinpath/bridge.checksum)"
                printf -v tuningchecksum "%s" "$(<$sourcebinpath/tuning.checksum)"
                printf "%s %s" "${bridgechecksum% *}" "${cni_mount_dir}/cnv-bridge" | sha256sum --check
                printf "%s %s" "${tuningchecksum% *}" "${cni_mount_dir}/cnv-tuning" | sha256sum --check

                ln -sf ${cni_mount_dir}/cnv-bridge ${cni_mount_dir}/bridge
                ln -sf ${cni_mount_dir}/cnv-tuning ${cni_mount_dir}/tuning
              fi
              echo 'Entering sleep... (success)'
              sleep infinity
```

Reply:
Ah you are right, the only difference would be that with 3 conditions we would not recreate the symlink, right?

I'm fine with the 2 condition approach.

Reply:
> 
> > I'm fine with the 2 condition approach.
> 
> so the PR as is right ? it only means that we won't deploy cnv-bridge on kubevirtci upstream atm, because bridge binary is there are we fine with it ? do we want to try and delete the binary ?

I'm fine with that.

Please make sure to test this change on Kubernetes and on OCP in various scenarios. I'm ok without all of them being automated, but let's at least try it manually.

Reply:
all looks fine

on CRC it has bridge binary and creates cnv-bridge sym link

on U/S it also has bridge binary and creates cnv-bridge sym link
and if i manually delete the bridge binary (and any cnv-bridge sym link leftovers from previous run)
it will deploy cnv bridge binary and create bridge symlink this time as expected